torch/_inductor/decomposition.py

```diff
@@ -144,8 +167,138 @@ def addmm(input, mat1, mat2, *, beta=1, alpha=1):
         if not isinstance(beta, numbers.Number) or beta != 1:
             input = input * beta
         return input + out
-    else:
-        return NotImplemented  # go directly to lowering
+
+    if (
+        config.shape_padding
+        and check_device_dtype(mat1, mat2)
+        and should_pad_bench(mat1, mat2, torch.ops.aten.addmm, input=input)
+    ):
+
+        k_padded_length = get_padded_length(mat1.shape[1])
+        n_padded_length = get_padded_length(mat2.shape[1])
+
+        if k_padded_length != 0:
+            mat1 = pad_dim(mat1, k_padded_length, 1)
+            mat2 = pad_dim(mat2, k_padded_length, 0)
+        if n_padded_length != 0:
+            if input is not None and input.is_cuda and input.dtype == torch.float32:
+                if input.dim() == 2:
+                    input = pad_dim(input, n_padded_length, 1)
+                elif input.dim() == 1:
+                    input = pad_dim(input, n_padded_length, 0)
+
+                mat2 = pad_dim(mat2, n_padded_length, 1)
+                return torch.ops.aten.addmm(input, mat1, mat2, beta=beta, alpha=alpha)[
+                    :, :-(n_padded_length)
+                ]
+        elif k_padded_length != 0:
+            return torch.ops.aten.addmm(input, mat1, mat2, beta=beta, alpha=alpha)
+
+    return NotImplemented  # go directly to lowering
+
+
+def bench(f, name=None, iters=100, warmup=5, display=False):
```

Comment:
Try to reuse the bench code in utils.py?
https://github.com/pytorch/pytorch/blob/master/torch/_inductor/utils.py#L106

Reply:
In `utils.py`, we add a new bench which is more accurate and reliable